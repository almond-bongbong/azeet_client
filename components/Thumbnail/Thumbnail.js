import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

const ThumbnailStyle = styled.div`
  & img {
      opacity: 0;
      transition: opacity .3s;
  }
  ${props => (!props.pending && css`
    & img {
      opacity: 1;
    }
  `)};
`;

const Thumbnail = ({ url }) => {
  const [pending, setPending] = useState(true);

  useEffect(() => {
    const image = new Image();

    if (url) {
      image.onload = () => setPending(false);
      image.src = url;
    }

    return () => {
      image.onload = null;
    };
  }, [url]);

  return (
    <ThumbnailStyle pending={pending ? 1 : 0}>
      {pending && <span>loading...</span>}
      <img src={url} alt="this is one" />
    </ThumbnailStyle>
  );
};

Thumbnail.propTypes = {
  url: PropTypes.string,
};

Thumbnail.defaultProps = {
  url: undefined,
};

export default Thumbnail;
