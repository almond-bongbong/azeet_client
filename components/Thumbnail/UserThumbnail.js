import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const UserThumbnailStyle = styled.div`
  overflow: hidden;
  width: 40px;
  height: 40px;
  border-radius: 50%;
  
  & img {
    width: 100%;
  }
`;

const UserThumbnail = ({ url }) => (
  <UserThumbnailStyle>
    <img src={url} alt="썸네일" />
  </UserThumbnailStyle>
);

UserThumbnail.propTypes = {
  url: PropTypes.string.isRequired,
};

export default UserThumbnail;
