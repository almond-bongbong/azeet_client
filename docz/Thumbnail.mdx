---
name: Thumbnail
menu: Image
---

import { useState } from 'react';
import { Playground } from 'docz'
import Thumbnail from 'components/Thumbnail/Thumbnail'

# Thumbnail

## Basic usage

<Playground>
  <Thumbnail url="https://placeimg.com/600/300/any" />
</Playground>
