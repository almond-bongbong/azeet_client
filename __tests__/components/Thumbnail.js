import React from 'react';
import { render } from '@testing-library/react';
import Thumbnail from 'components/Thumbnail/Thumbnail';

describe('Thumbnail', () => {
  it('matches snapshot', () => {
    const utils = render(<Thumbnail url="https://placeimg.com/600/300/any" />);
    expect(utils.container).toMatchSnapshot();
  });

  it('load image', async () => {
    const { queryByText, getByAltText } = render(<Thumbnail url="https://placeimg.com/100/100/any" />);
    const loading = queryByText('loading...');
    const img = getByAltText('this is one');
    expect(loading).toBeDefined();
    expect(img).toHaveStyle('opacity: 0');
  });
});
