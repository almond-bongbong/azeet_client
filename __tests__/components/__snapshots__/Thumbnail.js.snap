// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail matches snapshot 1`] = `
<div>
  <div
    class="Thumbnail__ThumbnailStyle-sc-1jpigpn-0 kUqCku"
  >
    <span>
      loading...
    </span>
    <img
      alt="this is one"
      src="https://placeimg.com/600/300/any"
    />
  </div>
</div>
`;
